### PR TITLE
feat: handle pending and incomplete positions

### PR DESCRIPTION
**Pull request by kaladinlight**
## Description

- Symmetric positions can be left in an incomplete state for various reasons. Add logic to detect incomplete positions and update the UI to allow for completion (add liquidity) and cancel (remove liquidity) flows
- Funds can be held in `pending` state within a position. Show pending funds in the remove liquidity flow so they can be removed if the user wishes

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

https://github.com/shapeshift/web/issues/6456

Med - want to make sure this allows resolution of position state and doesn't make things worse

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Ensure you are walked through the forced completion of an incomplete position via add liquidity on the position page
- Ensure you can withdraw (cancel) pending funds from a position. There may be various network reasons for funds to get stuck in pending, but incomplete positions are the easiest way to test and validate this

### Engineering

:point_up:

### Operations

:point_up:

## Screenshots (if applicable)

Incomplete position view:
image
image

Forced completion of the position from `Add Liquidity` view:
image
image

Withdrawal/Cancellation still supported via `Remove Liquidity` view:
image

### CODE REVIEW

File: src/assets/translations/en/main.json [hunk 1]
```diff
@@ -2230,6 +2231,8 @@
     "initialPricesAndPoolShare": "Initial Prices and Pool Share",
     "pricePerAsset": "%{from} per %{to}",
     "symAlert": "Your %{from} will be split into a 50:50 exposure to both %{to} and %{from} upon entering the pool.",
+    "incompletePositionDepositAlert": "This position is incomplete, please deposit the corresponding amount of %{asset} to finalize this position.",
+    "incompletePositionWithdrawAlert": "This position is incomplete, please cancel the position by removing 100% of the liquidity. You can also finalize the position via 'Add Liquidity'.",
```
Comment (kaladinlight) [inline src/assets/translations/en/main.json:2235]: Placeholder copy for now, up to refinement

File: src/lib/mixpanel/types.ts [hunk 2]
```diff
@@ -60,6 +60,9 @@ export enum MixPanelEvent {
   LpWithdrawInitiated = 'LP Withdraw Initiated',
   LpWithdrawSuccess = 'LP Withdraw Success',
   LpWithdrawFailed = 'LP Withdraw Failed',
+  LpIncompleteDepositConfirm = 'LP Incomplete Deposit Confirm',
+  LpIncompleteWithdrawPreview = 'LP Incomplete Withdraw Preview',
```
Comment (gomesalexandre) [inline src/lib/mixpanel/types.ts:64]: Tyvm 🙏🏽

File: src/pages/ThorChainLP/YourPositions.tsx [hunk 3]
```diff
@@ -93,6 +94,24 @@ export const YourPositions = () => {
...
+              )
+            }
+
+            if (position?.status.isPending) {
+              return (
+                <Tag colorScheme='yellow'>
+                  <Text translation='common.pending' />
+                </Tag>
+              )
+            }
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/YourPositions.tsx:112]: Can we return null here as a failsafe ? Or as an early return if `(!position)`

File: src/pages/ThorChainLP/components/AddLiquidity/AddLiquidity.tsx [hunk 4]
```diff
@@ -138,24 +138,31 @@ export const AddLiquidityRoutes: React.FC<AddLiquidityRoutesProps> = ({
...
-          }}
-        />
-      ) : null,
-    [confirmedQuote, history, mixpanel],
-  )
+  const renderAddLiquiditySweep = useCallback(() => {
+    if (!confirmedQuote) return null
+
+    const handleSweepSeen = () => {
+      if (confirmedQuote.positionStatus?.incomplete) {
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquidity/AddLiquidity.tsx:145]: 🧠

File: src/pages/ThorChainLP/YourPositions.tsx (hunk 3 above)
Comment (kaladinlight) [reply src/pages/ThorChainLP/YourPositions.tsx:112]: https://github.com/shapeshift/web/pull/6481/commits/8ee8e29363ff61d25d76cdca43c91085fdad40a6

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 5]
```diff
@@ -253,6 +253,17 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
     },
   })
 
+  // manages incomplete sym deposits by setting the already confirmed transaction as complete
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:256]: 🧠

**Review by gomesalexandre — COMMENTED** [review]
LGTM conceptually pending runtime testing of this

**Review by gomesalexandre — APPROVED** [review]
Tested locally:

### EVM

#### Incomplete position deposit

Screenshot 2024-03-21 at 12 41 52
Screenshot 2024-03-21 at 12 42 07
Screenshot 2024-03-21 at 12 44 50

- Position is correctly detected as incomplete with the option to deposit the missing side, or withdraw

Screenshot 2024-03-21 at 12 45 17
Screenshot 2024-03-21 at 12 45 32
Screenshot 2024-03-21 at 12 45 45

- Deposit of missing side

Screenshot 2024-03-21 at 12 45 58
image

- Deposit is now detected as complete

image

#### Incomplete position withdraw

- Incomplete deposit

image
image

- Deposit is detected as incomplete

image

- Withdraw of incomplete deposit

 
image
image
image
image

Confirmed withdraw is succesful and position is gone 🎉 

### UTXO

#### Incomplete position deposit

image

Note, full DOGE balance sent to next receive index before continuing to the next step, to ensure we still sweep.

- Position is correctly detected as incomplete with the option to deposit the missing side, or withdraw

image


- Deposit of missing side

image
image

Sweep was correctly detected as needed 🎉 

image
image
image

- Deposit is now detected as complete

image

**Pull request merged**